paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -142,6 +145,74 @@ public boolean fileExists(Path path) {
         }
     }
 
+    public void mergeBranch(String branchName) {
+        checkArgument(
+                !branchName.equals(DEFAULT_MAIN_BRANCH),
+                "Branch name '%s' do not use in merge branch.",
+                branchName);
+        checkArgument(branchExists(branchName), "Branch name '%s' doesn't exist.", branchName);
+        Long earliestSnapshotId = snapshotManager.earliestSnapshotId(branchName);
+        Snapshot earliestSnapshot = snapshotManager.snapshot(branchName, earliestSnapshotId);
+        long earliestSchemaId = earliestSnapshot.schemaId();
+
+        try {
+            // Delete snapshot, schema, and tag from the main branch which occurs after
+            // earliestSnapshotId
+            List<Path> deleteSnapshotPaths =
```

Comment:
When deleting, consider whether to also delete the LASTEST file? If the main branch keeps writing, the latest snapshot generated will be deleted during merge, which may cause the snapshot of the LAST written by the main branch to not be found?

Reply:
@sunxiaojian I think we should document that stop writing to main branch before merging branch.

Reply:
> @sunxiaojian I think we should document that stop writing to main branch before merging branch.

@schnappi17 If it does not affect the original business, there may be a situation where both the main branch and the branch are writing. Although the time may be short, it is best to delete the LATEST file as this will not affect the business

Reply:
> @sunxiaojian I think we should document that stop writing to main branch before merging branch.

What we need to ensure is to stop writing to the main branch when merging to it. For example, the main branch creates "snapshot-1", "snapshot-2", "snapshot-3", and creates a "tag-2" based on "snapshot-2". At this time, the snapshot ID recorded in the LATEST file is 3. At this time, we create a branch "branch-2" based on "tag-2". When merging, we first delete the "snapshot-3" file of the main branch. At this time, I think LASTEST should also be deleted synchronously, otherwise we need to update the LASTEST file when copying branch files. Obviously, deleting directly is the best method because LASTEST supports lazy loading

Reply:
> @sunxiaojian I think we should document that stop writing to main branch before merging branch.

What we need to ensure is to stop writing to the main branch when merging to it. For example, the main branch creates "snapshot-1", "snapshot-2", "snapshot-3", and creates a "tag-2" based on "snapshot-2". At this time, the snapshot ID recorded in the LATEST file is 3. At this time, we create a branch "branch-2" based on "tag-2". When merging, we first delete the "snapshot-3" file of the main branch. At this time, I think LASTEST should also be deleted synchronously, otherwise we need to update the LASTEST file when copying branch files. Obviously, deleting directly is the best method because LASTEST supports lazy loading

Reply:
> @sunxiaojian I think we should document that stop writing to main branch before merging branch.

What we need to ensure is to stop writing to the main branch when merging to it. For example, the main branch creates "snapshot-1", "snapshot-2", "snapshot-3", and creates a "tag-2" based on "snapshot-2". At this time, the snapshot ID recorded in the LATEST file is 3. At this time, we create a branch "branch-2" based on "tag-2". When merging, we first delete the "snapshot-3" file of the main branch. At this time, I think LASTEST should also be deleted synchronously, otherwise we need to update the LASTEST file when copying branch files. Obviously, deleting directly is the best method because LASTEST supports lazy loading